src/widget/wmainmenubar.cpp

```diff
@@ -0,0 +1,692 @@
+    QAction* pHelpManual = new QAction(manualTitle, this);
+    pHelpManual->setStatusTip(manualText);
+    pHelpManual->setWhatsThis(buildWhatsThis(manualTitle, manualText));
+    m_visitUrlMapper.setMapping(pHelpManual, qManualUrl.toString());
+    connect(pHelpManual, SIGNAL(triggered()), &m_visitUrlMapper, SLOT(map()));
+    pHelpMenu->addAction(pHelpManual);
+
+    QString shortcutsTitle = tr("&Keyboard Shortcuts") + externalLinkSuffix;
+    QString shortcutsText = tr("Speed up your workflow with keyboard shortcuts.");
+    QAction* pHelpShortcuts = new QAction(shortcutsTitle, this);
+    pHelpShortcuts->setStatusTip(shortcutsText);
+    pHelpShortcuts->setWhatsThis(buildWhatsThis(shortcutsTitle, shortcutsText));
+    m_visitUrlMapper.setMapping(pHelpShortcuts, MIXXX_SHORTCUTS_URL);
+    connect(pHelpShortcuts, SIGNAL(triggered()), &m_visitUrlMapper, SLOT(map()));
+    pHelpMenu->addAction(pHelpShortcuts);
+
+    QString feedbackTitle = tr("Send Us &Feedback") + externalLinkSuffix;
+    QString feedbackText = tr("Send feedback to the Mixxx team.");
+    QAction* pHelpFeedback = new QAction(feedbackTitle, this);
+    pHelpFeedback->setStatusTip(feedbackText);
+    pHelpFeedback->setWhatsThis(buildWhatsThis(feedbackTitle, feedbackText));
+    m_visitUrlMapper.setMapping(pHelpFeedback, MIXXX_FEEDBACK_URL);
+    connect(pHelpFeedback, SIGNAL(triggered()), &m_visitUrlMapper, SLOT(map()));
+    pHelpMenu->addAction(pHelpFeedback);
+
+    QString translateTitle = tr("&Translate This Application") + externalLinkSuffix;
+    QString translateText = tr("Help translate this application into your language.");
+    QAction* pHelpTranslation = new QAction(translateTitle, this);
+    pHelpTranslation->setStatusTip(translateText);
+    pHelpTranslation->setWhatsThis(buildWhatsThis(translateTitle, translateText));
+    m_visitUrlMapper.setMapping(pHelpTranslation, MIXXX_TRANSLATION_URL);
+    connect(pHelpTranslation, SIGNAL(triggered()), &m_visitUrlMapper, SLOT(map()));
+    pHelpMenu->addAction(pHelpTranslation);
+
+    pHelpMenu->addSeparator();
+
+    QString aboutTitle = tr("&About");
+    QString aboutText = tr("About the application");
+    QAction* pHelpAboutApp = new QAction(aboutTitle, this);
+    pHelpAboutApp->setStatusTip(aboutText);
+    pHelpAboutApp->setWhatsThis(buildWhatsThis(aboutTitle, aboutText));
+    pHelpAboutApp->setMenuRole(QAction::AboutRole);
+    connect(pHelpAboutApp, SIGNAL(triggered()),
+            this, SIGNAL(showAbout()));
+
+    pHelpMenu->addAction(pHelpAboutApp);
+    addMenu(pHelpMenu);
+}
+
+void WMainMenuBar::onLibraryScanStarted() {
+    emit(internalLibraryScanActive(true));
+}
+
+void WMainMenuBar::onLibraryScanFinished() {
+    emit(internalLibraryScanActive(false));
+}
+
+void WMainMenuBar::onNewSkinLoaded() {
+    // Delete all the old VisibilityControlConnections.
+    qDeleteAll(m_viewActionConnections);
+
+    for (auto it = m_viewActions.constBegin(); it != m_viewActions.constEnd(); ++it) {
```

Comment:
This is the only place where we read m_viewActions. So this should be a QList of a new struct containing
```
QActions* pAction 
ConfigKey configKey
```
This way we get around the anonymous access below and the performance will be improved.